### ISSUE TITLE
Include python 3.8 interpreter.

### ISSUE THREAD

**Issue by crunk1** [GoogleCloudPlatform/python-runtime#225]
Any chance we can get the shiny, new Python version?

**Comment by tylervick** [comment]
Yes please - 3.8 has been out for months and even GAE standard supports this already.

**Comment by VelizarVESSELINOV** [comment]
It is good that 3.8 is out, but still python google-appengine docker image don't recognize Python 3.8
```
Step 1/7 : FROM gcr.io/google-appengine/python
latest: Pulling from google-appengine/python
Status: Downloaded newer image for gcr.io/google-appengine/python:latest
Step 2/7 : RUN virtualenv -p python3.8 /env
The path python3.8 (from --python=python3.8) does not exist
The command '/bin/sh -c virtualenv -p python3.8 /env' returned a non-zero code: 3
ERROR
ERROR: build step 0 "gcr.io/cloud-builders/docker" failed: exit status 3
```

**Comment by dhendry** [comment]
We are half way to 3.9 already, it would be great to have 3.8 support!

**Comment by psztur** [comment]
come oooon :) an ETA please?

**Comment by psztur** [comment]
I am really keen on seeing newer version available soon.  Python3.8 will bring tremendous perfomance improvement over 3.7 version.  Is there something I can help you with?  Some issues you are having that need additional investigation?  Some testing that needs to be done?

**Comment by VelizarVESSELINOV** [comment]
Just a reminder to the Google support team we are in May 2022, and you are at least three years late. The last 3 Python versions are unsupported by Google Python App Engine Flex, and Google Python App Engine Flex supports four obsolete versions.

Python 2.7 is the default, and the latest Python version supported is 3.7.

Version | Released | Security Support | Release
-- | -- | -- | --
3.10 | 7 months ago (04 Oct 2021) | Ends in 4 years (04 Oct 2026) | 3.10.4
3.9 | 1 year and 6 months ago (05 Oct 2020) | Ends in 3 years and 5 months (05 Oct 2025) | 3.9.12
3.8 | 2 years and 6 months ago (14 Oct 2019) | Ends in 2 years and 5 months (14 Oct 2024) | 3.8.13
3.7 | 3 years and 10 months ago (27 Jun 2018) | Ends in 1 year and 1 month (27 Jun 2023) | 3.7.13
3.6 | 5 years ago (23 Dec 2016) | Ended 4 months ago (23 Dec 2021) | 3.6.15
3.5 | 6 years ago (13 Sep 2015) | Ended 1 year and 7 months ago (13 Sep 2020) | 3.5.10
3.4 | 8 years ago (16 Mar 2014) | Ended 3 years ago (18 Mar 2019) | 3.4.10
3.3 | 9 years ago (29 Sep 2012) | Ended 4 years and 7 months ago (29 Sep 2017) | 3.3.7
2.7 | 11 years ago (03 Jul 2010) | Ended 2 years and 4 months ago (01 Jan 2020) | 2.7.18

https://endoflife.date/python

**Comment by VelizarVESSELINOV** [comment]
Python 3.9 is available as a preview if you request by filling the form on: https://issuetracker.google.com/issues/172647331 👍

**Comment by yedpodtrzitko** [comment]
any update on this please?

**Comment by nmartin-sens** [comment]
Do you know if Google plans to have Python 3.8+ in App Engine Flex? It would be really great.

**Comment by dsanmart** [comment]
If you really need support for newer python versions, I recommend you consider using a custom runtime and build your own docker image importing base image from https://hub.docker.com/_/python. This would look something like:

**app.yaml**
```yaml
runtime: custom
env: flex
```

**Dockerfile**
```go
FROM python:3.10

ENV VIRTUAL_ENV /env
ENV PATH /env/bin:$PATH

WORKDIR /app

ADD requirements.txt .
RUN pip install -r requirements.txt 

ADD . /app

EXPOSE 8080

ENTRYPOINT [ "streamlit", "run", "app.py", "--server.port", "8080" ]
```
